src/main/scala/com/fulcrumgenomics/bam/api/SamOrder.scala

```diff
@@ -176,11 +176,11 @@ object SamOrder {
     override val subSort:    Option[String] = Some("template-coordinate")
     override val sortkey: SamRecord => A = rec => {
       val readChrom = if (rec.unmapped)     Int.MaxValue else rec.refIndex
-      val mateChrom = if (rec.mateUnmapped) Int.MaxValue else rec.mateRefIndex
+      val mateChrom = if (!rec.paired || rec.mateUnmapped) Int.MaxValue else rec.mateRefIndex
```

Comment:
@nh13 What do you think about adding `@inline final def unpaired: Boolean = !paired` to `SamRecord`? I feel like it would be tidier than using `!rec.paired` everywhere.